src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -2002,7 +2002,7 @@ namespace nupic {
 	  << "Invalid index: " << *it
 	  << " - Should be >= 0 and < number of columns:" << nCols();
       
-      for (size_type i = 1; i < (size_type)(end - begin); ++i)
+      for (auto i = 1; i < (end - begin); ++i)
```

Comment:
Specifically, use this to replace the entire check that we are discussing. It should do exactly the same thing but without doing any iterator arithmetic.